yarn-project/aztec-nr/compressed-string/src/compressed_string.nr

```diff
@@ -0,0 +1,139 @@
+use dep::aztec::types::type_serialization::TypeSerializationInterface;
+use dep::protocol_types::utils::field::field_from_bytes;
+use dep::std;
+
+// A Fixedsize Compressed String.
+// Essentially a special version of Compressed String for practical use.
+struct FixedsizeCompressedString{
+  value: Field
+}
+
+impl FixedsizeCompressedString{
+  pub fn is_eq(self, other: FixedsizeCompressedString) -> bool {
+    self.value == other.value
+  }
+
+  pub fn from_field(input_field: Field) -> Self {
+    Self {value: input_field}
+  }
+
+  pub fn from_string(input_string: str<31>) -> Self {
+    Self {value: field_from_bytes(input_string.as_bytes(), true)}
+  }
+
+  pub fn to_bytes(self) -> [u8; 31] {
```

Comment:
for my understanding - why 31 and not 32?

Reply:
32 * 8 = 256. We don't have that many bits to use in a field.